src/t_zset.c

```diff
@@ -511,6 +551,21 @@ unsigned long zslGetRank(zskiplist *zsl, double score, sds ele) {
     return 0;
 }
 
+/* Find the rank for an element by both score and key.
+ * Returns 0 when the element cannot be found, rank otherwise.
+ * Note that the rank is 1-based due to the span of zsl->header to the
+ * first element. */
+unsigned long zslGetRankByNode(zskiplist *zsl, zskiplistNode *x) {
```

Comment:
What do you mean by "Returns 0 when the element cannot be found"? The element is provided by the caller in `x`, isn't it?

It returns 0 only if `x` is the last element in the skiplist?

Reply:
Sorry - bad copy-and-paste :) let me fix that